### ISSUE TITLE
Error: Failure when receiving data from the peer

### ISSUE THREAD

**Issue by blasp** [randyzwitch/RSiteCatalyst#174]
I have been getting this error when using RSiteCatalyst with a shiny app. I have to mention that the application works great on my machine but as soon as I deploy the application one of the ggplot charts or DT datatables using data extracted wiht RSiteCatalyst returns this message "Error: Failure when receiving data from the peer"

I am completely out of my deph as I do not even know to what to look for when troubleshooting this error.

The query I am running is quite a simple one

QueueTrended("mysuite",
                 date.from= format(Sys.Date()-90, format="%Y-%m-%d"), 
                 date.to= format(Sys.Date()-1, format="%Y-%m-%d"),
                 date.granularity = "day", 
                 metrics=c("mymetric"),
                 elements= c("myelement"),
                 selected = c("xxx"),
                 segment.inline=  myinlineselectedSegment,
                 interval.seconds = 5)

**Comment by randyzwitch** [comment]
Hard to say, but are you getting this error before you get a DataFrame with your data? 

The only I can think of if it works on your machine but not remotely would be that the remote machine doesn't have Internet access or has a firewall that only lets certain sites through (I.e. Just allows CRAN). 

Where is this app being hosted, an internal server or a public cloud like Amazon?

**Comment by blasp** [comment]
Thanks Randy - the incoming data querried with RsiteCatalyst is assigned to a data frame. So really the code looks like this

SiteCatData <- QueueTrended("mysuite",
date.from= format(Sys.Date()-90, format="%Y-%m-%d"), 
date.to= format(Sys.Date()-1, format="%Y-%m-%d"),
date.granularity = "day", 
metrics=c("mymetric"),
elements= c("myelement"),
selected = c("xxx"),
segment.inline= myinlineselectedSegment,
interval.seconds = 5)

The app is hosted in shiny standard plan.

**Comment by randyzwitch** [comment]
Do you actually get data from the API, or does the failure happen before you get a DataFrame?

**Comment by blasp** [comment]
I am not sure how to check if the data is being received. What is a way to get a delivery confirmation ?

**Comment by randyzwitch** [comment]
I would probably use tryCatch, but honestly, I'm not that familiar with Shiny nor Shiny apps hosting

http://mazamascience.com/WorkingWithData/?p=912

**Comment by slin30** [comment]
I have no idea if this will help, but one thing I've noticed with receiving datetime data, e.g. from QueueTrended, is that I consistently have issues with the POSIXlt format that is returned in a data.frame by default. I first noticed that I was not able to sort (in the RStudio viewer, or via View()) on the datetime column from a freshly pulled QueueTime report, and when converting to a data.table (via as.data.table), I'd go from e.g. 300 rows to 11. 

Apologies for not having time to provide a proper response with details, but as a quick test, it might be worthwhile to try converting the datetime to character, and, if you like, back to datetime via e.g. as.Date (I use the as.Date in zoo).
